### PR TITLE
[master] Optimize Refresh bootmap script 

### PR DESCRIPTION
**Pull request by changzhi1990**
Optimize Refresh bootmap script #1433 and #1431

This patch does the following things:
1.  Parse 'lsluns' command output if not specific wwpns in command line
2.  More details info output
3. Change the return info, use "WWPNs" instead of "Physical WWPN"
4.  Fix "rd.zfcp=" error. Add "0x" prefix to each wwpn in "rd.zfcp=" option
5.  Optimize "disconnectFCP" function

### CODE REVIEW

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 1]
```diff
@@ -317,35 +315,99 @@ function refreshFCPBootmap {
...
+    fi
   done
 
   # Remove duplicates
   wwpns=($(echo "${wwpns[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
 
   # Try to connect FCP
-  if [[ ! wwpns[0] ]]; then
+  if [[ ! $wwpns[0] ]]; then
     printError "wwpns[0] is empty. Please check the parameters in command line."
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: we should not expose such wwpns[0] in the error msg.
Change this msg to "No available wwpn found."

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 2]
```diff
@@ -317,35 +315,99 @@ function refreshFCPBootmap {
...
-  if (( $? )); then
-    printError "Failed to connect disk: ccw-0.0.${fcps[0]}-0x${wwpns[0]}:${lun}."
-    exit 10
-  fi
-  refreshZIPL
+  for i in "${!fcps[@]}"
+  do
+    for j in "${!wwpns[@]}"
+    do
+        if [[ "$i" -eq "$j" ]];then
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: not sure what this means...

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 3]
```diff
@@ -317,35 +315,99 @@ function refreshFCPBootmap {
...
+        fi
+    done
+  done
+
+  # Delete files which in /dev/disk/by-path/ccw-0.0.*
+  for block in "${block_devs[@]}"
+  do
+    realpath=`readlink -f $block`
+    sdxname=`echo "$realpath" | cut -d "/" -f 3`
+    if [[ -e /sys/block/$sdxname/device/delete ]];then
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: duplicate with disconnect.

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 4]
```diff
@@ -317,35 +315,99 @@ function refreshFCPBootmap {
...
-    done
+    # Try to connect FCP device and online it.
+    /opt/zthin/bin/smcli Image_Device_Dedicate -T $zthinUserID -v $fcp -r $fcp 0 > /dev/null
+    /sbin/cio_ignore -r $fcp > /dev/null
+    chccwdev -e $fcp
+    host=`lszfcp | grep "$fcp" | cut -d " " -f 2`
+    echo "- - -" > /sys/class/scsi_host/$host/scan
+    # Use sleep command to make sure all files are generated
+    # by chccwdev command.
+    sleep 1
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: need to enhance this sleep 1. digg more when you got time.

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 5]
```diff
@@ -317,35 +315,99 @@ function refreshFCPBootmap {
...
+    done
+  done
+
+  refreshZIPL ${fcps[0]} ${wwpns[0]} ${lun}
+
+  for i in "${!fcps[@]}"
+  do
+    for j in "${!wwpns[@]}"
+    do
+        if [[ "$i" -eq "$j" ]];then
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: not only disconnectFCP when i==j. Should be called for i*j times.

File: zthin-parts/zthin/bin/refresh_bootmap (hunk 3 above)
Comment (changzhi1990) [reply zthin-parts/zthin/bin/refresh_bootmap]: Use these commands to remove files immediately

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 6]
```diff
@@ -335,17 +381,34 @@ function refreshFCPBootmap {
   wwpns=($(echo "${wwpns[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap]: maybe this line should be move to before line 368?

**Comment by dongyanyang** [comment]
@jichenjc @bjhuangr please help to merge this. Thanks!

**Comment by jichenjc** [comment]
I need take some deep look into this, before that ,can you fix the merge conflict?

**Pull request merged**